internal/versions/versions.go

```diff
@@ -36,29 +36,99 @@ func SupportedK8sVersions() []string {
 	return validVersionsSorted
 }
 
+// SupportedValidK8sVersions returns a typed list of supported Kubernetes versions.
+func SupportedValidK8sVersions() (res []ValidK8sVersion) {
+	for _, v := range SupportedK8sVersions() {
+		res = append(res, ValidK8sVersion(v))
+	}
+	return
+}
```

Comment:
I don't think that we should implement a function here which is only used in test and which simply converts the type. This is quite confusing as `SupportedK8sVersions()` already outputs only supported versions.  
Another way would be to replace the `SupportedK8sVersions()` function with `SupportedValidK8sVersions()`

Reply:
What about making this a private function in *_test.go?

Reply:
@derpsteb  i did that now :)

Reply:
@3u13r can u have another look?